cmake/DetectOptions.cmake

```diff
@@ -170,8 +170,27 @@ endif()
 
 set(mpi_find_components C)
 
-# Cuda
-if(ADIOS2_USE_CUDA)
+# Kokkos
+if(ADIOS2_USE_Kokkos)
+  if(ADIOS2_USE_Kokkos STREQUAL AUTO)
+    find_package(Kokkos 3.7 QUIET)
```

Comment:
@anagainaru we have to hold on this. Spock/Crusher does not offer Kokkos 3.7 just yet. I am in another conversation where I have just asked this, when will Kokkos 3.7 be avaiable for those machine. I will keep you in the loop.

Reply:
This is needed to be able to use the same device id for the Kokkos backend as for the application. If the CUDA application sets the device id not to the first GPU in the list, there will be errors between the backend and CUDA. It's not an issue for now that Kokkos 3.7 is not on Crusher, you can install your own until they add it.